pilot/pkg/xds/endpoints/endpoint_builder.go

```diff
@@ -661,7 +661,8 @@ func buildEnvoyLbEndpoint(b *EndpointBuilder, e *model.IstioEndpoint, mtlsEnable
 	tunnel := supportTunnel(b, e)
 	// Only send HBONE if its necessary. If they support legacy mTLS and do not explicitly PreferHBONE, we will use legacy mTLS.
 	// However, waypoints (TrafficDirectionInboundVIP) do not support legacy mTLS, so do not allow opting out of that case.
-	if mtlsEnabled && !features.PreferHBONESend && b.dir != model.TrafficDirectionInboundVIP {
```

Comment:
what happened to `mtlsEnabled`?

Reply:
So consider a ISTIO_MUTUAL rule and a ztunnel destination.

Before, we get mtlsEnabled and set tunnel=false and don't use HBONE/mTLS.

What we want is "if ISTIO_MUTUAL use HBONE or mTLS, preferring mTLS if possible". So we disable TLS only if the backend supportsMtls

Reply:
Ack, makes sense!